tests/unit/test_configuration.py

```diff
@@ -555,18 +572,30 @@ def test_no_noise() -> None:
     # where all noise levels are 0.0
     no_noise_config = get_configuration("no_noise")
 
-    for dataset in no_noise_config.keys():
-        dataset_dict: LayeredConfigTree = no_noise_config[dataset]
-        dataset_row_noise_dict: LayeredConfigTree = dataset_dict[Keys.ROW_NOISE]
-        dataset_column_dict: LayeredConfigTree = dataset_dict[Keys.COLUMN_NOISE]
-        for row_noise_type in dataset_row_noise_dict.keys():
-            noise_specific_config: LayeredConfigTree = dataset_row_noise_dict[row_noise_type]
-            for key, value in noise_specific_config.items():
-                assert dataset_row_noise_dict[row_noise_type][key] == 0.0
-        for column in dataset_column_dict.keys():
-            column_noise_dict: LayeredConfigTree = dataset_column_dict[column]
-            for column_noise_type in column_noise_dict.keys():
-                assert column_noise_dict[column_noise_type][Keys.CELL_PROBABILITY] == 0.0
+    dataset_names = [dataset.name for dataset in DATASET_SCHEMAS]
+    for dataset in dataset_names:
+        dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset)
+        for row_noise_type in dataset_schema.row_noise_types:
+            parameters = []
+            if row_noise_type.probability is not None:
+                parameters.append("row_probability")
```

Comment:
Nit: it's safer to do this:
```suggestion
                parameters.append(row_noise_type.probability_key)
```